reflectometry/amor_reduction_detailed.ipynb

```diff
@@ -0,0 +1,1106 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab315161",
+   "metadata": {},
+   "source": [
+    "- $\\sigma \\theta/\\theta$: finally, this accounts for the width of the beam on the sample or the size of the sample (which ever is smaller), the FWHM is the range of possible $\\theta$-values, accounting for the gravity correction discussed above and the sample/beam geometry,\n",
+    "\n",
+    "$$\n",
+    "\\frac{\\sigma\\theta}{\\theta} = \\frac{1}{2\\sqrt{2\\ln2}}\\frac{\\theta_{\\text{max}} - \\theta_{\\text{min}}}{\\theta_{\\text{mid}}}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "centered-eligibility",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "beam_size = 0.001 * sc.units.m\n",
+    "sample_size = 0.5 * sc.units.m\n",
+    "beam_on_sample = beam_size / sc.sin(data.bins.coords['theta'])\n",
+    "half_beam_on_sample = (beam_on_sample / 2.0)\n",
+    "offset_positive = sc.geometry.position(\n",
+    "    data.coords[\"sample_position\"].fields.x, \n",
+    "    data.coords[\"sample_position\"].fields.y, \n",
+    "    data.coords[\"sample_position\"].fields.z + half_beam_on_sample)\n",
+    "offset_negative = sc.geometry.position(\n",
+    "    data.coords[\"sample_position\"].fields.x, \n",
+    "    data.coords[\"sample_position\"].fields.y, \n",
+    "    data.coords[\"sample_position\"].fields.z - half_beam_on_sample)\n",
+    "y_measured = data.coords[\"position\"].fields.y\n",
+    "z_measured = data.coords[\"position\"].fields.z\n",
+    "z_origin = offset_positive.fields.z\n",
+    "y_origin = offset_positive.fields.y\n",
+    "y_dash = y_dash0(data.bins.coords[\"velocity\"], z_origin, y_origin, z_measured, y_measured)\n",
+    "intercept = y_origin - y_dash * z_origin\n",
+    "y_true = z_measured * y_dash + intercept\n",
+    "angle_max = sc.to_unit(sc.atan(y_true / z_measured), 'deg')\n",
+    "z_origin = offset_negative.fields.z\n",
+    "y_origin = offset_negative.fields.y\n",
+    "y_dash = y_dash0(data.bins.coords[\"velocity\"], z_origin, y_origin, z_measured, y_measured)\n",
+    "intercept = y_origin - y_dash * z_origin\n",
+    "y_true = z_measured * y_dash + intercept\n",
+    "angle_min = sc.to_unit(sc.atan(y_true / z_measured), 'deg')\n",
+    "fwhm_to_std = 2 * np.sqrt(2 * np.log(2))\n",
+    "sigma_theta_position = (angle_max - angle_min) / fwhm_to_std\n",
+    "data.attrs[\"sigma_theta_by_theta\"] = sigma_theta_position / data.bins.coords['theta']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f317df7a",
+   "metadata": {},
+   "source": [
+    "These three contributors to the resolution function are combined to give the total resolution function, \n",
+    "\n",
+    "$$\n",
+    "\\frac{\\sigma q_z}{q_z} = \\sqrt{\\bigg(\\frac{\\sigma\\lambda}{\\lambda}\\bigg)^2+\\bigg(\\frac{\\sigma\\gamma}{\\theta}\\bigg)^2+\\bigg(\\frac{\\sigma\\theta}{\\theta}\\bigg)^2}\n",
```

Comment:
Looks like the norm of a vector, but I suppose storing the three components as a single vector is not quite working for you?

Reply:
It is just incidental that there are three components, these three are the largest contributors to the resolution function but in theory there could be many more. I think that using a vector would make it more confusing for the reader.